moderntribe/square-one
# Properly prefix commits with their Jira ticket

## What does this do/fix?

- Looks like I broke this some time ago, and different bash versions/shells have been a problem in the past. This converts the functionality to use PHP which everyone should have on their system so if you have a branch checked out with a Jira ticket in the name, it will automatically prefix it with `[JIRA-123]: original commit message` as long as you have git hooks enabled.
- Fixes lefthook to 0.7.5 until 0.8.0 is released due to performance issues with yarn and npx: https://github.com/evilmartians/lefthook/issues/241 (it's like 1000% slower for me).

## QA

- `nvm use`
- `yarn install`
- `lefthook install -a`
- `git checkout -b feature/SQONE-112/my-feature`
- Modify some files...
- `git add .`
- `git commit -m 'add some cool feature'`
- Checking your git commits you'll see: `[SQONE-112]: add some cool feature`

## Tests

Does this have tests?

- [x] No, this doesn't need tests because...well it could use some tests.



[JIRA-123]: https://moderntribe.atlassian.net/browse/JIRA-123?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[SQONE-112]: https://moderntribe.atlassian.net/browse/SQONE-112?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
